tests/test_web_sendfile_functional.py

```diff
@@ -288,13 +308,12 @@ async def handler(request):
     client = await aiohttp_client(app)
 
     resp = await client.get("/")
-    assert 200 == resp.status
-    body = await resp.read()
-    assert b"hello aiohttp\n" == body
-    ct = resp.headers["CONTENT-TYPE"]
-    assert "text/plain" == ct
-    encoding = resp.headers["CONTENT-ENCODING"]
-    assert "gzip" == encoding
+    assert resp.status == 200
+    assert (
+        resp.headers["Content-Type"],
```

Comment:
Please, make this multiple assertions, each clearly making one simple check. Complex checks are hard to parse in the test output and in code.

Reply:
Sorry, I disagree here, and think `pytest` does a good job of comparing sequences item by item.

I made that change specifically to aid me in debugging the issue with 3.8, and it made the issue very clear.  Asserting one at a time makes sense if a previous one failing guarantees the rest will fail (e.g. if the response status is not okay, then the body and headers won't exist).  But the other 3 were wrong and asserting one at a time meant I could only know one value at a time (the first one), which isn't very useful.  It was originally asserting the body first, which is probably the last one that would go wrong.  I shouldn't have to change a test multiple times to get a clear picture of the results in the log.

Reply:
That's understandable. But still, complex assertions of semantically different values are discouraged. Pytest can compare them but the test run output is a mess. Plus it's super ugly in code. A simple comparison mustn't be multiline, because it increases cognitive complexity for no good reason.

Reply:
> That's understandable. But still, complex assertions of semantically different values are discouraged.

Can you point out somewhere that is documented for my reading and edification?  I did some `pytest` researching before I made that change and found recommendations to organize assertions similarly to deal with the issue, so I'd like to read the opposing views.

> Pytest can compare them but the test run output is a mess.

Here's the output that made the problem clear to me: https://github.com/aio-libs/aiohttp/actions/runs/7683502126/job/20939089887

What do you feel is a mess there?

> Plus it's super ugly in code. A simple comparison mustn't be multiline, because it increases cognitive complexity for no good reason.

It seems not much different than any other conditional that gets formatted to multiple lines, for which I'm sure there are countless occurrences?  And are you saying the reason I just gave is not a good one?  Couldn't I just argue it's less complex with less assertion steps: (1) response is okay, (2) actual response = expected response?

I'm not trying to be difficult here - I'm just genuinely perplexed by your request.  I don't think I've ever been asked to make something objectively less useful in order to purely make the code subjectively prettier.

Reply:
In any case, in the interest of avoiding an infinite review loop here, I made the change, but put them in an order that makes some sense: content encoding, then type, and finally body.  I hope that resolves this.

Reply:
> What do you feel is a mess there?

That one is short enough, but it would hurt consistency if we were to use different styles depending on each output example, which might even change over time.

Reply:
> Can you point out somewhere that is documented for my reading and edification? I did some `pytest` researching before I made that change and found recommendations to organize assertions similarly to deal with the issue, so I'd like to read the opposing views.

Unfortunately, I don't have anything like that. Perhaps it's tribal knowledge in some sense. But I did see complex chunks of output at some point, that would be more complicated to understand. Also note, that people's perception may differ due to their experience and tooling used, and dev setup. And chasing inclusivity, I'm trying to use practices that would lean towards accessibility. For example, I use far larger fonts than many people — it helps to take some strain off of my eyes. This has a side effect of rendering weirdly and generating more line breaks, especially in various split views like diff pages. This informs a choice of shorter lines in the source code. But it is also something that sometimes has a similar effect in log outputs.

This is one of my main motivations to reduce cognitive complexity in text, wherever I see it. I hope this provides you with a bit more insight.

Reply:
@webknjaz thanks for answering my questions in detail, and for the record...

> And chasing inclusivity, I'm trying to use practices that would lean towards accessibility.

Cheers to that as you're talking to a fellow a11y enthusiast (expert if I'm feeling boastful) and legally blind developer. 🥂 👨🏻‍🦯 

In any case, the changes here are now minimal with regard to assertions following #8136, so I'll tag this resolved.

Reply:
Thank you for your insights too!